README.md

```diff
@@ -1,109 +1,142 @@
 ## ocaml-git -- Git format and protocol in pure OCaml
 
-Support for on-disk and in-memory Git stores. Can read and write all
-the Git objects: the usual blobs, trees, commits and tags but also
-the pack files, pack indexes and the index file (where the staging area
-lives).
+Support for on-disk and in-memory Git stores. Can read and write all the Git
+objects: the usual blobs, trees, commits and tags but also the pack files, pack
+indexes and the index file (where the staging area lives - only for `git-unix`
+package).
+
+All the objects share a consistent API, and convenience functions are provided
+to manipulate the different objects. For instance, it is possible to make a pack
+file position independent (as the Zlib compression might change the relative
+offsets between the packed objects), to generate pack indexes from pack files,
+or to expand the filesystem of a given commit.
 
-All the objects share a consistent API, and convenience functions are
-provided to manipulate the different objects. For instance, it is
-possible to make a pack file position independent (as the Zlib
-compression might change the relative offsets between the packed
-objects), to generate pack indexes from pack files, or to expand
-the filesystem of a given commit.
+The library comes with some command-line tools called `ogit-*` as a
+Proof-of-concept of the core library which shares a similar interface with
+`git`, but where all operations are mapped to the API exposed `ocaml-git` (and
+hence using only OCaml code). However, these tools are not done to be use. They
+are just examples of how to use `ocaml-git`.
 
-The library comes with some command-line tools called `ogit-*` as a Proof-of-concept
-of the core library which shares a similar interface with `git`, but where all
-operations are mapped to the API exposed `ocaml-git` (and hence using only OCaml
-code). However, these tools are not done to be use. They are just examples of how to use `ocaml-git`.
+`ocaml-git` wants to be a low-level library for [irmin][irmin]. By this fact,
+high-level processes such as a ([patience][patience-diff]) diff, `git status`,
+etc. are not implemented.
 
-The API documentation is available
-[online](http://mirage.github.io/ocaml-git/).
+The API documentation is available [online][documentation].
 
 [![Build Status](https://travis-ci.org/mirage/ocaml-git.svg?branch=master)](https://travis-ci.org/mirage/ocaml-git)
 
-### Build and Install Instructions
+### Build, Install Instructions and Packages
 
 To build and install the project, simply run:
-```
+```sh
 $ opam install git
+$ opam install git-unix
 ```
 
+As a MirageOS project, `ocaml-git` is system agnostic. However, it provides a
+`git-unix` package which uses UNIX _syscall_ and it able to introspect a Git
+repository as you usually know. However, `ocaml-git` handles only Git objects
+and it does not _populate_ your filesystem as `git` does.
```

Comment:
This is a super helpful clarification. I'd even suggest moving this caveat up to the top of the readme, as it makes it very clear that this library is not suitable for many use cases which a user might initial expect from a library called "git".